File: backend/HolaSmileDMS/HDMS_API/Domain/Entities/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class Task
{
    [Key]
    public int TaskID { get; set; }

    [ForeignKey("Assistant")]
    public int? AssistantID { get; set; }
    public Assistant? Assistant { get; set; }

    [ForeignKey("TreatmentProgress")]
    public int? TreatmentProgressID { get; set; }
    public TreatmentProgress? TreatmentProgress { get; set; }

    [MaxLength(200)]
    public string? ProgressName { get; set; }

    [MaxLength(500)]
    public string? Description { get; set; }

    [MaxLength(50)]
    public string? Status { get; set; }

    public TimeSpan? StartTime { get; set; }
    public TimeSpan? EndTime { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public int? CreatedBy { get; set; }
    public int? UpdatedBy { get; set; }
}

File: frontend/src/types/task.ts
export interface BasicTask {
    taskId: number
    progressName: string
    description: string
    status: boolean
    startTime: string
    endTime: string
    assistantId?: number
    assistantName?: string
}

export interface TaskAssignment {
    assistantId: number
    treatmentProgressID: number
    progressName: string
    status: boolean
    description: string
    startTime: string
    endTime: string
}

export interface TaskFilter {
  status: string
  timeRange: string
  dentist: string
  procedure: string
}
export type TaskStatus = "Pending" | "Completed"

export interface Task {
  taskId: number
  progressName: string
  description: string
  status: TaskStatus
  startTime: string
  endTime: string
  treatmentProgressId: number
  treatmentRecordId: number
  treatmentDate: string
  procedureName: string
  dentistName: string
  symptoms: string
  diagnosis: string
}

export interface AssignedTask extends BasicTask { }
